python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -124,14 +131,32 @@ def _check_repo_has_partition_set(
         raise DagsterBackfillFailedError(
             f"Could not find partition set {partition_set_name} in repository {repo_name}. "
         )
+    return external_repo.get_external_partition_set(partition_set_name)
+
+
+def _subdivide_partition_key_range(
+    partitions_def: PartitionsDefinition,
+    partition_key_range: PartitionKeyRange,
+    instance: DagsterInstance,
+    max_range_size: Optional[int],
+):
+    if max_range_size is None:
+        return [partition_key_range]
+    else:
+        keys = partitions_def.get_partition_keys_in_range(
+            partition_key_range, dynamic_partitions_store=instance
+        )
+        chunks = [keys[i : i + max_range_size] for i in range(0, len(keys), max_range_size)]
+        return [PartitionKeyRange(start=chunk[0], end=chunk[-1]) for chunk in chunks]
 
 
 def _get_partitions_chunk(
     instance: DagsterInstance,
     logger: logging.Logger,
     backfill_job: PartitionBackfill,
     chunk_size: int,
-) -> Tuple[Sequence[str], str, bool]:
+    partition_set: ExternalPartitionSet,
```

Comment:
would be a big fan of a rename to RemotePartitionSet while you are at it in this stack

Reply:
Have got that in my queue. Big stack of renames I will land this week.